### PR TITLE
Cart-Drawer Functionality

### PR DESCRIPTION
**Pull request by shradhdhadeardigital**
1) Implemented dynamic cart-drawer.
2) Added products to the cart drawer on the "Add to Cart" button.
3) Enabled cart add, update, and remove functionalities.

### CODE REVIEW

File: assets/cart-drawer.js [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+    const cartUrl = `${rootUrl}/?sections=cart-drawer`;
+    let cartHTML;
+
+    if (!updatedSections) {
+      const fetchedSections = await fetchAndParseCartAndHeader(cartUrl);
+      cartHTML = fetchedSections.cartHTML;
+    } else {
+      const parser = new DOMParser();
+      const cartHtmlDoc = parser.parseFromString(updatedSections['cart-drawer'], 'text/html');
+      cartHTML = cartHtmlDoc.querySelector('.shopify-section-yg-cart-drawer .drawer-cart').innerHTML;
```
Comment (mihir-bombay-studio) [inline assets/cart-drawer.js:391]: This class shopify-section-yg-cart-drawer is for yummygums theme

File: assets/cart-drawer.js [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+    _learnq.push([
+      'track',
+      'Added to Cart',
+      {
+        total_price: custom_cart_total / 100,
+        $value: custom_cart_total / 100,
+        items: cart.items,
+      },
+    ]);
+  }
```
Comment (mihir-bombay-studio) [inline assets/cart-drawer.js:354]: _learnq is for klaviyo we don't require it in our theme

File: assets/cart-drawer.js [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+  let updatedSections = sections;
+  return { cart: updatedCart, sections: updatedSections };
+}
+
+function findLineForGratisProduct(cart, variantId) {
+  const index = cart.items.findIndex(
+    (item) => item.properties._gratis_product === 'true' && item.variant_id === parseInt(variantId)
+  );
+  return index + 1;
+}
```
Comment (mihir-bombay-studio) [inline assets/cart-drawer.js:335]: These custom functions are for another theme we don't require these here yet

File: assets/cart-drawer.js [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+
+function toggleCartVisibility(open) {
+  const OVERFLOW_HIDDEN = 'overflow-hidden';
+  const action = open ? 'add' : 'remove';
+  document.querySelector('.drawer-cart')?.classList[action]('drawn');
+  document.querySelector('.cart-overlay')?.classList[action]('drawn');
+  document.body.classList[action](OVERFLOW_HIDDEN);
+}
+
+window.upsellOpenedByDrawer = false;
```
Comment (mihir-bombay-studio) [inline assets/cart-drawer.js:287]: We don't have upsell drawer

File: assets/cart-drawer.js
Comment (mihir-bombay-studio) [inline assets/cart-drawer.js:1]: Need to clean up the custom code before using it in our theme

**Review by mihir-bombay-studio — CHANGES_REQUESTED** [review]
Can you also create a loom demonstrating this new feature

File: assets/cart-drawer.js (hunk 1 above)
Comment (shradhdhadeardigital) [reply assets/cart-drawer.js:391]: I already used this class in the liquid file >> https://prnt.sc/X5QyX-89aN3x
But now I have given class names related to our theme.

File: assets/cart-drawer.js (hunk 2 above)
Comment (shradhdhadeardigital) [reply assets/cart-drawer.js]: I have removed this code.

File: assets/cart-drawer.js (hunk 3 above)
Comment (shradhdhadeardigital) [reply assets/cart-drawer.js]: I have updated this code.

File: assets/cart-drawer.js (hunk 4 above)
Comment (shradhdhadeardigital) [reply assets/cart-drawer.js]: I have removed this code.

**Pull request merged**